app/services/scheduler_maintenance.py

```
@@ -87,41 +97,62 @@ async def _report_incomplete_run(result: HistoricalRefreshResult) -> None:
+            await _report_incomplete_run(result)
+            if not result.can_advance_freshness:
+                return
+    except Exception as e:
+        logger.error("Error refreshing historical results: %s", e, exc_info=True)
+        return
+
+    try:
+        await get_database().set_cache_meta(
+            _HISTORICAL_REFRESH_META_KEY, datetime.now(timezone.utc).isoformat()
+        )
+    except Exception as e:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)